### ISSUE TITLE
Phone app missing permission for secret codes

### ISSUE THREAD

**Issue by losuler** [RattlesnakeOS/microg#5]
## Background
There's currently a bug with microG where Google/Firebase Cloud Messaging doesn't connect (in settings shows as `Current state: Disconnected`). [There is a way to fix it](https://github.com/microg/android_packages_apps_GmsCore/wiki/Helpful-Information) which involves typing the secret code `*#*#2432546#*#*` to do a manual device registration or running the command `adb shell am broadcast [...]`.

## Issue
The issue in Android 10 is that there is now [restrictions on starting activities from the background](https://developer.android.com/guide/components/activities/background-starts) and the Phone app built with AOSP lacks the permission `START_ACTIVITIES_FROM_BACKGROUND` required to run secret codes.

When you attempt to run any secret code the Phone app simply crashes and the following error comes up in `logcat`:

```
java.lang.SecurityException: Permission Denial: android.provider.Telephony.SECRET_CODE broadcast from com.android.phone (pid=1810, uid=1001) requires android.permission.START_ACTIVITIES_FROM_BACKGROUND
```

This permissions issue in the AOSP Phone app also effects every other secret code such as `*#*#4636#*#*`, which among other things, allows you to lock to 4G only (`Preferred network type` in settings can't do this).

The `adb shell am broadcast [...]` command also cannot be run also in AOSP/RattlesnakeOS as it requires `root`. Unless you also patched this in.

## Possible solutions
Unless this is fixed in microG (has been outstanding for about a year), I assume this may be able to be fixed in AOSP if the Phone app is patched to include the permission `START_ACTIVITIES_FROM_BACKGROUND`. However I don't know enough about AOSP to know how to create a patch like this.

## Note
I just thought I'd mention this here because it's likely someone else will come across this problem.

**Comment by pgera** [comment]
I have done the gcm registration process with the dialer code a few times on Android 10 now. I didn't get a crash, and gcm did register after that. What I have noticed though is that gcm does not register on its own in Android 10. That is, the dialer code is always needed on the first time setup, whereas it used to register automatically in earlier versions. There are also other issues with location access in Android 10 (https://github.com/microg/android_packages_apps_GmsCore/issues/986).

**Comment by losuler** [comment]
> I have done the gcm registration process with the dialer code a few times on Android 10 now.

Are you able to run other commands in the Phone app like `*#*#4636#*#*`? What version of the Phone app are you using?

**Comment by pgera** [comment]
Yes, I get the testing and statistics screen with that code. Phone app's version is 23. This is on sargo with the latest build.

**Comment by losuler** [comment]
The Phone app is also `version 23.0` for me and I'm on the latest build. However this on `walleye`. The only way I could be sure is if I flashed a factory image of RattlesnakeOS again. Backing up everything is a bit of a pain.
